alacritty_terminal/src/window.rs

```diff
@@ -408,6 +386,16 @@ impl Window {
         }
     }
 
+    pub fn set_maximized(&self, maximized: bool) {
+        #[cfg(not(any(target_os = "macos", windows)))]
+        {
+            let window = self.window();
+            if self.event_loop.is_x11() {
+                window.set_maximized(maximized);
+            }
+        }
+    }
```

Comment:
The way this method is split apart now is a bit dangerous. The only reason why things are maximized after starting on X11 is because doing it before showing doesn't work, not because `set_maximized` only works on X11. This changes the window API in a way that makes it seem like `set_maximized` only works on X11, which is not the case.

A better way to do this would be to always call `window.set_maximized(bool)` in the Window's method and handle the logic at the callsite in display.rs.